### PR TITLE
Feature/cvsb 2868

### PR DESCRIPTION
**Pull request by acozianu**

### CODE REVIEW

File: src/models/TestStationDAO.js [hunk 1]
```diff
@@ -13,6 +13,22 @@ class TestStationDAOmock {
     return dbClient.scan({ TableName: this.tableName }).promise()
   }
 
+  getTestStationEmailByPNumber (testStationPNumber) {
```
Comment (KHodderDVSA) [inline src/models/TestStationDAO.js]: No directly related to a change you made, but this class is "TestStationDAOmock" - not correct.

File: tests/unit/testStationsFunction.unitTest.js [hunk 2]
```diff
@@ -1,9 +1,32 @@
...
 
 describe('getTestStations', () => {
   it('should return a promise', () => {
     return LambdaTester(GetTestStationsFunction)
       .expectResolve()
   })
 })
+
+describe('getTestStationsEmail', () => {
+  it('should return an error', () => {
```
Comment (KHodderDVSA) [inline tests/unit/testStationsFunction.unitTest.js]: These tests descriptions could really use some more context e.g.
"should return an error when sending no parameters"

File: tests/unit/testStationsFunction.unitTest.js [hunk 3]
```diff
@@ -1,9 +1,32 @@
...
+  })
+  it('should return a promise', () => {
+    return LambdaTester(GetTestStationsEmailFunction.getTestStationEmails).event({
+      pathParameters: {
+        testStationPNumber: '87-1369569' }
+    })
+      .expectResolve()
+  })
+  it('should return a promise', () => {
+    return LambdaTester(GetTestStationsEmailFunction.getTestStationEmails).event({
```
Comment (KHodderDVSA) [inline tests/unit/testStationsFunction.unitTest.js]: Does this test add anything? I believe that it isn't expected to return anything, but that isn't actually tested

File: src/functions/getTestStationEmails.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  const testStationDAO = new TestStationDAO()
+  const service = new TestStationService(testStationDAO)
+  const testStationPNumber = event.pathParameters.testStationPNumber
+
+  return service.getTestStationEmails(testStationPNumber)
+    .then((data) => {
+      return new HTTPResponse(200, data)
+    })
+    .catch((error) => {
+      return new HTTPResponse(error.statusCode, error.body)
```
[the remark this answers is not in the document]
Comment (acozianu) [reply src/functions/getTestStationEmails.js]: The HTTPError extends Error and is throwable.
We used the HTTPResponse to send data to FE. Event is ok or not, we send something as a response

File: src/models/TestStationDAO.js (hunk 1 above)
Comment (acozianu) [reply src/models/TestStationDAO.js]: Done

File: tests/unit/testStationsFunction.unitTest.js (hunk 2 above)
Comment (acozianu) [reply tests/unit/testStationsFunction.unitTest.js]: Done

File: tests/unit/testStationsFunction.unitTest.js (hunk 3 above)
Comment (acozianu) [reply tests/unit/testStationsFunction.unitTest.js]: last time I checked, it does

File: src/functions/getTestStationEmails.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (KHodderDVSA) [reply src/functions/getTestStationEmails.js]: I think we should be able to return bodies with thrown errors, but a discussion for another time perhaps.

**Pull request merged**